File: algorithms/1_easy/F/03438_find_valid_pair_of_adjacent_digits_in_string/solution.py
class Solution:
    def findValidPair(self, s: str) -> str:
        freqs = [0] * 10
        n = len(s)

        for ch in s:
            freqs[ord(ch) - ord('0')] += 1

        for i in range(n - 1):
            if (
                s[i] != s[i + 1] and
                ord(s[i]) - ord('0') == freqs[ord(s[i]) - ord('0')] and
                ord(s[i + 1]) - ord('0') == freqs[ord(s[i + 1]) - ord('0')]
            ):
                return s[i] + s[i + 1]

        return ""

File: algorithms/1_easy/F/03438_find_valid_pair_of_adjacent_digits_in_string/solution.cpp
class Solution {
public:
    string findValidPair(string s) {
        array<int, 10> freqs{};
        const size_t len{s.size()};

        for (const char& ch : s) {
            freqs[static_cast<size_t>(ch - '0')] += 1;
        }

        for (size_t i{0}; i < len - 1; ++i) {
            if (
                s[i] != s[i + 1] &&
                s[i] - '0' == freqs[static_cast<size_t>(s[i] - '0')] &&
                s[i + 1] - '0' == freqs[static_cast<size_t>(s[i + 1] - '0')]
            ) {
                return s.substr(i, 2);
            }
        }

        return "";
    }
};
